chore: skip CI on draft PR

amitmiran137 opened the pull request:
### SUMMARY
It feels like lately CI workflows wait for hours before triggering next to release due-date when many PR's are active.
We should push the contributers to start Draft PR's which reduce the amount of  'ready to merge' PRs active in parallel.
this will enable the remaining group to run in iterations much more faster 

solution:
When a PR is a draft CI would skip
every additional commit(push) would still skip CI
when PR is changed to **ready for review** this would trigger the workflow again and CI would run

once PR gets merged CI should just run again based on push event to the master branch

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
<!--- Skip this if not applicable -->

### TEST PLAN
<!--- What steps should be taken to verify the changes -->

### ADDITIONAL INFORMATION
<!--- Check any relevant boxes with "x" -->
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->
- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

Commit:
chore: test how CI behaves when running a draft PR

Commit:
chore: disable other workflows

Commit:
chore: additional disabling

ktmud commented:
Nice. This will definitely help the overall CI speed! Thanks for tackling this.

ktmud commented on .github/workflows/docker.yml:
```diff
@@ -9,6 +9,7 @@ on:
 
 jobs:
   build:
+    if: github.event.pull_request.draft == false
```
Will this affect "push" runs as well?

amitmiran137 replied:
it shouldn't conflict bc we have the 
on: 
   push
   
which triggers every time but the condition for enabling the job is always **when not draft** 
so only when you push and PR is not draft CI will run

Commit:
fix: by default pull_request triggers only on
synchronize, opened, reopened
I've added ready_for_review for the CI to trigger again when moving from draft to ready_for_review

ktmud reviewed (approved):
LGTM

Commit:
fix: by default pull_request triggers only on
synchronize, opened, reopened
I've added ready_for_review for the CI to trigger again when moving from draft to ready_for_review
